File: scala/src/main/scala/com/webtrekk/datastreams/example/config/ConfigKeys.scala
package com.webtrekk.datastreams.example.config

object ConfigKeys {
  val ClientId = "client-id"
  val GroupId = "consumer-group"
  val Endpoints = "endpoints"
  val AutoOffsetResetPolicy = "auto-offset-reset-policy"

  val SecurityProtocol = "security-protocol"
  val SecuritySaslMechanism = "security-sasl-mechanism"

  val SecurityScramUsername = "username"
  val SecurityScramPassword = "password"

  val Topic = "topic"

  object Streams {
    val NumOfThreads = "streams-num-of-threads"
    val DeserializationExceptionHandler = "streams-deserialization-error-handler"
  }
}

File: java/src/main/java/com/webtrekk/datastreams/examples/config/ConfigKeys.java
package com.webtrekk.datastreams.examples.config;

public class ConfigKeys {

    public static final String ClientId = "client-id";
    public static final String GroupId = "consumer-group";
    public static final String Endpoints = "endpoints";
    public static final String AutoOffsetResetPolicy = "auto-offset-reset-policy";

    public static final String SecurityProtocol = "security-protocol";
    public static final String SecuritySaslMechanism = "security-sasl-mechanism";

    public static final String SecurityScramUsername = "username";
    public static final String SecurityScramPassword = "password";

    public static final String Topic = "topic";

    public static class Streams {
        public static final String NumOfThreads = "streams-num-of-threads";
        public static final String DeserializationExceptionHandler = "streams-deserialization-error-handler";
    }

}
